sale_invoice_plan/models/sale.py

```diff
@@ -45,14 +45,13 @@ def _compute_invoice_plan_process(self):
         for rec in self:
             has_invoice_plan = rec.use_invoice_plan and rec.invoice_plan_ids
             to_invoice = rec.invoice_plan_ids.filtered(lambda l: not l.invoiced)
-            if rec.state == "sale" and has_invoice_plan and to_invoice:
-                if rec.invoice_status == "to invoice" or (
-                    rec.invoice_status == "no"
-                    and "advance" in to_invoice.mapped("invoice_type")
-                ):
-                    rec.invoice_plan_process = True
-                    continue
-            rec.invoice_plan_process = False
+            inv_or_adv = rec.invoice_status == "to invoice" or (
```

Comment:
@dreispt just FYI, I saw this part of code that you try to refactor on your code is incorrect. This might be the reason that advance is not working?
I try to refactor here too, this will remain the same logic and it works for me.